ribs/archives/_archive_base.py

```diff
@@ -466,7 +504,33 @@ def add(self,
         measures_batch_can = measures_batch[can_insert]
         index_batch_can = index_batch[can_insert]
         metadata_batch_can = metadata_batch[can_insert]
-        old_objective_batch_can = old_objective_batch[can_insert]
+        old_objective_batch_can = old_objective_arr[can_insert]
+
+        def groupby(a, b):
```

Comment:
Talk to me about this in person -- I don't understand what's going on here, but I think we may be able to do it faster with the `numpy_groupies` library.